Unit Test

MinJiang00 opened the pull request.

MinJiang00 committed:
test runs, more fields needed to be test

MinJiang00 committed:
test runs, more fields needed to be test

MinJiang00 committed:
tests done, except for kerberos

MinJiang00 committed:
tests done, except for kerberos

MinJiang00 committed:
keys done, dns waiting

MinJiang00 committed:
keys done, dns waiting

mathewab commented on dns_config/test/api_acl_test.go:
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
+	"io"
+	"net/http"
 	"testing"
+)
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
+type RoundTripFunc func(req *http.Request) *http.Response
 
-	openapiclient "github.com/infobloxopen/bloxone-go-client/dns_config"
-	"github.com/infobloxopen/bloxone-go-client/internal"
-)
+func (f RoundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
 
-func Test_dns_config_AclAPIService(t *testing.T) {
+func NewTestClient(fn RoundTripFunc) *http.Client {
+	return &http.Client{
+		Transport: RoundTripFunc(fn),
+	}
+}
```
can you move this to `internal/utils`?

mathewab commented on dns_config/test/api_acl_test.go:
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
 	t.Run("Test AclAPIService AclCreate", func(t *testing.T) {
 
-		t.Skip("skip test") // remove to run test
+		//t.Skip("skip test") // remove to run test
```
can you please remove commented code?

mathewab commented on dns_config/test/api_auth_nsg_test.go:
```diff
@@ -1,93 +1,237 @@
 DNS Configuration API
 
-Testing AuthNsgAPIService
+Testing AclAPIService
```
Shouldn't that be as it was - `AuthNsgAPIService`?

mathewab commented on dns_config/test/api_acl_test.go:
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
+			}
+		})
+
+		configuration := internal.NewConfiguration()
+		configuration.HTTPClient = testClient
+
+		aclAPI := dns_config.NewAPIClient(configuration)
+		ctx := context.Background()
```
you could have a common function for this in each package. 

```
func newAPITestClient(testFn RoundTripFunc) *dns_config.APIClient {
    configuration := internal.NewConfiguration()
    configuration.HTTPClient = NewTestClient(testFn)

   return dns_config.NewAPIClient(configuration)
}
```
That way you can avoid some duplication and simply write 

```
dnsConfigAPI := newAPITestClient(func(req *http.Request) *http.Response {
	require.Equal(t, "PATCH", req.Method)
        .....
}

mathewab commented on dns_config/test/api_acl_test.go:
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
-		require.NotNil(t, resp)
-		assert.Equal(t, 200, httpRes.StatusCode)
+		testClient := NewTestClient(func(req *http.Request) *http.Response {
+			require.Equal(t, "POST", req.Method)
```
use the constant from `http` package for all methods and status 

```suggestion
			require.Equal(t, http.MethodPost, req.Method)
```

mathewab commented on dns_config/test/api_acl_test.go:
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
 
-		t.Skip("skip test") // remove to run test
+			return &http.Response{
+				StatusCode: 200,
```
```suggestion
				StatusCode: http.StatusOk,
```

mathewab commented on dns_config/test/api_acl_test.go:
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
+		//t.Skip("skip test") // remove to run test
 
-		resp, httpRes, err := apiClient.AclAPI.AclCreate(context.Background()).Execute()
+		dummyAcl := dns_config.ConfigACL{
+			Comment: openapiclient.PtrString("This is a dummy ACL for testing."),
+			Id:      openapiclient.PtrString("dummyAclId"),
+			Name:    "dummyAclName",
+		}
```
Put all object definitions that are reused outside the Test function.

mathewab commented on keys/api_tsig.go:
```diff
@@ -13,6 +13,7 @@ package keys
 import (
 	"bytes"
 	"context"
+	_ "github.com/infobloxopen/bloxone-go-client/dns_config"
```
why this change?

mathewab commented on keys/model_keys_tsig_key.go:
```diff
@@ -38,6 +39,7 @@ type KeysTSIGKey struct {
 	Tags map[string]interface{} `json:"tags,omitempty"`
 	// Time when the object has been updated. Equals to _created_at_ if not updated after creation.
 	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+	res       interface{}
```
why did you add this?

venkat-iblox commented on keys/.openapi-generator/VERSION:
```diff
@@ -1 +1 @@
-7.0.1
+7.2.0
```
let's stick to using generator 7.0.1 for now

MinJiang00 committed:
Complete unit tests for dns_data, dns_config, and keys modules

MinJiang00 committed:
Complete unit tests for dns_data, dns_config, and keys modules

MinJiang00 committed:
Just ran make client

MinJiang00 committed:
ran go fmt ./...

MinJiang00 committed:
Merge branch 'tsig-keys-test' of https://github.com/MinJiang00/bloxone-go-client into tsig-keys-test

MinJiang00 committed:
Tests

MinJiang00 committed:
Unit Tests For Keys, DNS With Generator Working at version 7.3.0

- Unit tests for keys and dns completed, with generator upgraded to 7.3.0
- Ran go fmt ./..., but the equivalence make fmt failed

MinJiang00 committed:
Merge branch '7.3.0' into tsig-keys-test

MinJiang00 committed:
Ran 'go fmt ./...' for formatting check and investigated version 7.3.0

MinJiang00 committed:
Merge branch 'tsig-keys-test' of https://github.com/MinJiang00/bloxone-go-client into unit-test

MinJiang00 committed:
Complete unit tests for ipam, dhcp, dns_data, dns_config, and keys modules

MinJiang00 committed:
Complete unit tests for ipam, dhcp, dns_data, dns_config, and keys modules

mathewab commented on dns_config/test/api_acl_test.go:
```diff
@@ -7,87 +7,158 @@ Testing AclAPIService
-	apiClient := openapiclient.NewAPIClient(configuration)
-
-	t.Run("Test AclAPIService AclCreate", func(t *testing.T) {
+var dns_config_Post = openapiclient.ConfigACL{
```
variable names should be camelCase

mathewab commented on dns_config/test/api_acl_test.go:
```diff
@@ -7,87 +7,158 @@ Testing AclAPIService
+}
 
-		resp, httpRes, err := apiClient.AclAPI.AclCreate(context.Background()).Execute()
+func Test_dns_config_AclAPIService(t *testing.T) {
```
You can remove the package name from the tests

```suggestion
func TestAclAPIService(t *testing.T) {
```

MinJiang00 committed:
Complete unit tests for dns_data, dns_config, and keys modules:
1. Removed the first line in test
2. Updated test object name to camel case

mathewab commented on dns_config/test/api_auth_zone_test.go:
```diff
@@ -5,101 +5,187 @@ Testing AuthZoneAPIService
 
-func Test_dns_config_AuthZoneAPIService(t *testing.T) {
+var ConfigCopyAuthZonePost = openapiclient.ConfigCopyAuthZone{
+	Comment: openapiclient.PtrString("This is a copyping AuthZone for testing."),
```
`copyping` typo?

MinJiang00 committed:
Complete unit tests for dns_data, dns_config, ipam/dhcp, infra_mgmt and keys modules:
1. Removed the first line in test
2. Updated test object name to camel case

mathewab commented on ipam/test/api_dhcp_host_test.go:
```diff
@@ -7,76 +7,127 @@ Testing DhcpHostAPIService
 
 // Code generated by OpenAPI Generator (https://openapi-generator.tech);
```
This line is not removed in `ipam`

MinJiang00 committed:
Complete unit tests for dns_data, dns_config, ipam/dhcp, infra_mgmt and keys modules:
1. Removed the first line in test
2. Updated test object name to camel case
